libraries/Microsoft.Bot.StreamingExtensions/PayloadStream.cs

```diff
@@ -0,0 +1,179 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.StreamingExtensions.Payloads;
+
+namespace Microsoft.Bot.StreamingExtensions
+{
+    internal class PayloadStream : Stream
+    {
+        private readonly ContentStreamAssembler _assembler;
+        private readonly Queue<byte[]> _bufferQueue = new Queue<byte[]>();
+
+        private readonly SemaphoreSlim dataAvailable = new SemaphoreSlim(0, int.MaxValue);
+        private readonly object syncLock = new object();
+        private long _producerLength = 0;       // total length
+        private long _consumerPosition = 0;     // read position
+
+        private byte[] _active = null;
+        private int _activeOffset = 0;
```

Comment:
This member variable is *only* ever accessed in Read and ReadAsync. I can't figure out who is consuming it. 

Can it be removed?